Cards

Daniil9596 opened the pull request:
Add card base logic (create, view)

Daniil9596 commented on pom.xml:
```diff
@@ -170,6 +170,9 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.21.0</version>
+                <configuration>
+                    <useSystemClassLoader>false</useSystemClassLoader>
```
A few days ago I updated my linux packages (jdk8 was on the list). Then compilation fails. I read that it because Debian package bug (https://stackoverflow.com/questions/53010200/maven-surefire-could-not-find-forkedbooter-class). In the StackOverflow I found solution to insert that <configuration> and all works fine now for me.

Meisssshu replied:
I hope it won't affect your teammates

Daniil9596 commented on src/main/java/com/epam/lab/paymentsystem/controller/CardController.java:
```diff
@@ -10,18 +14,25 @@
 public class CardController {
+  @Autowired private UserService userService;
+  @Autowired private CardService cardService;
+  @Autowired private AccountService accountService;
```
Yes, accountService is redundant.

Daniil9596 commented on src/main/java/com/epam/lab/paymentsystem/controller/CardController.java:
```diff
@@ -32,4 +43,40 @@ public String getAccountPage(@PathVariable(name = "accountId") long id, Model mo
+  @GetMapping(value = "/user/account/{accountId}/addCard")
+  public String getAddCardPage(Model model) {
+    LOGGER.info("Access to card creation page");
+    model.addAttribute("userList", userService.getAllUsers());
```
I use datalist in html form for selecting user - card holder (it works as select list with search).

Daniil9596 commented on src/main/java/com/epam/lab/paymentsystem/service/impl/CardServiceImpl.java:
```diff
@@ -1,22 +1,38 @@
-  @Autowired
-  private AccountService accountService;
+
+  private CardRepository cardRepository;
```
Yes, it was bad))

Daniil9596 commented on src/main/java/com/epam/lab/paymentsystem/service/impl/CardServiceImpl.java:
```diff
@@ -1,22 +1,38 @@
+
+  private static final Logger LOGGER = LogManager.getLogger(AccountServiceImpl.class);
+
+  @Autowired private UserService userService;
```
Ok, it appeared after merging. I fixed similar things.

Meisssshu commented on src/main/java/com/epam/lab/paymentsystem/service/impl/CardServiceImpl.java:
```diff
@@ -15,31 +15,53 @@
+   */
   @Override
   public List<Card> getAllCardsByLogin(String login) {
     return null;
```
Is it supposed to be so?

Daniil9596 replied:
No, I have just added implementation and test))

nikandtos commented on src/main/java/com/epam/lab/paymentsystem/controller/CardController.java:
```diff
@@ -32,4 +45,40 @@ public String getAccountPage(@PathVariable(name = "accountId") long id, Model mo
+      @RequestParam(name = "login") String userLogin,
+      Model model) {
+
+    CardDto cardDto = new CardDto(0, accountId, userLogin, cardLabel, false);
```
you can create cardDto with help of thymeleaf and take it as argument in controller

Daniil9596 replied:
Done:
1) Added dto (new CardDto()) attribute to model in getAddCardPage() method.
2) Replace 2 params with 1 dto via @ModelAttribute annotation.

nikandtos commented on src/main/webapp/WEB-INF/views/addCard.html:
```diff
@@ -0,0 +1,46 @@
+    <a href="?lang=en" class="btn btn-link active" role="button" aria-pressed="true">EN</a>
+    <a href="?lang=ru" class="btn btn-link active" role="button" aria-pressed="true">RU</a>
+</div>
+<form id="registrationLayer" action="addCard" method="post">
```
`<form id="registrationLayer" action="addCard" method="post" th:object="${cardDto}">`

Daniil9596 replied:
Thanks, it works)

nikandtos commented on src/main/webapp/WEB-INF/views/addCard.html:
```diff
@@ -0,0 +1,46 @@
+        <tr>
+            <td><label for="login"><b><p th:text="#{user.login}"/></b></label></td>
+            <td><input list="logins" name="login" class="form-control input-lg"
+                       th:placeholder="#{user.placeholder.name}" id="login" required>
```
`<input list="logins" name="login" class="form-control input-lg" th:field="*{login}"
                       th:placeholder="#{user.placeholder.name}" id="login" required>`

Daniil9596 replied:
And it too)

Daniil9596 commented:
A few things is done:
1) Added exception to card creation.
2) Added test for exceptional case.
3) Added check if list of cards is empty in html + added raw header to cards table.

Daniil9596 commented:
Fixed @Autowired
